### PR TITLE
Fix warnings related to dangling comments and lossy conversions

### PR DESCRIPTION
**Pull request by jasonkatonica**
Recently numerous warnings have displayed related to lossy conversions and dangling `javadoc` comments.

This update ensures that there are no dangling `javadoc` comments these changes fall into various categories such as unused code that has `javadoc` comments embedded, empty `javadoc` comments, duplicate `javadoc` comments, or javadoc comments that are not located above a declaration.

Updates were also made to `XDHPublicKeyImpl`, `BaseTestChaCha20KAT`, and `BaseTestChaCha20NoReuse` to explicitly cast an int to a byte OR make use of a common utility function `BaseUtils.hexStringToByteArray` that does not have such a lossy conversion present.

While fixing dangling comments one test `testRSASignatureSHA512_256` was found to be commented out. It appears to execute correctly and does seem like a valid test so this test was uncommented.

Two spaces found in the pom.xml were removed for consistency.

Signed-off-by: Jason Katonica <katonica@us.ibm.com>

### CODE REVIEW

**Review by johnpeck-us-ibm — APPROVED** [review]

**Review by KostasTsiounis — CHANGES_REQUESTED** [review]

File: src/main/java/com/ibm/crypto/plus/provider/PSSParameters.java [hunk 1]
```diff
@@ -56,16 +56,7 @@
  { OID id-mgf1 PARAMETERS OAEP-PSSDigestAlgorithms },
  ...  -- Allows for future expansion --
  }
-
- *
- * </pre>
```
[the remark this answers is not in the document]
Comment (jasonkatonica) [reply src/main/java/com/ibm/crypto/plus/provider/PSSParameters.java:61]: Yes agreed when using html we should close the `<pre>` tag to be cleaner. I also noticed that the `javadoc` was missing leading `* `values on each line so i added those too in the latest commit https://github.com/IBM/OpenJCEPlus/pull/268/commits/8e3ffae5d89087120efef46f43d481510b57c423

File: src/test/java/ibm/jceplus/junit/base/BaseTestTruncatedDigest.java [hunk 2]
```diff
@@ -116,22 +116,17 @@ public void testRSASignatureSHA512_224() throws Exception {
...
-    // }
-
-
+    public void testRSASignatureSHA512_256() throws Exception {
+        PSSParameterSpec pssParameter = new PSSParameterSpec("SHA512/256", "MGF1",
+                MGF1ParameterSpec.SHA512_256,20, 1);
+        try {
+            dotestSignature(content, IBM_ALG, 2048, pssParameter, getProviderName());
+        } catch (Exception e) {
+            // TODO Auto-generated catch block
```
[the remark this answers is not in the document]
Comment (jasonkatonica) [reply src/test/java/ibm/jceplus/junit/base/BaseTestTruncatedDigest.java]: Yes I removed this comment and others exactly or very similar to this in commit https://github.com/IBM/OpenJCEPlus/pull/268/commits/8e3ffae5d89087120efef46f43d481510b57c423

**Review by KostasTsiounis — APPROVED** [review]
LGTM

**Comment by jasonkatonica** [comment]
x86_64_linux, ppc64le_linux, s390x_linux, x86_64_windows

**Review by taoliult — APPROVED** [review]
LGTM

**Pull request merged**